### PR TITLE
[autolinking] add packageName to android react-native-config

### PR DESCRIPTION
**Pull request by Kudo**
# Why

add missing `project.android.packageName` when integration 0.75 core autolinking with our implementation. it shows `RNGP - Autolinking: Could not find project.android.packageName in react-native config output! Could not autolink packages without this field.` error.

# How

add missing `project.android.packageName` 

# Test Plan

- add unit test
- test 0.75 core autolinking integration

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [x] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Comment by Kudo** [comment]
* **#30913** <a href="https://app.graphite.dev/github/pr/expo/expo/30913?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @Kudo and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/expo-modules-autolinking/src/reactNativeConfig/__tests__/reactNativeConfig-test.ts [hunk 1]
```diff
@@ -160,6 +162,42 @@ describe(findDependencyRootsAsync, () => {
   });
 });
 
+describe(resolveAppProjectConfigAsync, () => {
+  it('should return app project config for android', async () => {
```
Comment (tsapeta) [inline packages/expo-modules-autolinking/src/reactNativeConfig/__tests__/reactNativeConfig-test.ts:166]:
```suggestion
  it('should return app project config for Android', async () => {
```

File: packages/expo-modules-autolinking/src/reactNativeConfig/__tests__/reactNativeConfig-test.ts [hunk 2]
```diff
@@ -160,6 +162,42 @@ describe(findDependencyRootsAsync, () => {
...
+      {
+        "android": {
+          "packageName": "com.test",
+          "sourceDir": "/app/android",
+        },
+      }
+    `);
+  });
+
+  it('should return app project config for ios', async () => {
```
Comment (tsapeta) [inline packages/expo-modules-autolinking/src/reactNativeConfig/__tests__/reactNativeConfig-test.ts:189]:
```suggestion
  it('should return app project config for iOS', async () => {
```

**Review by tsapeta — APPROVED** [review]

File: packages/expo-modules-autolinking/src/reactNativeConfig/__tests__/reactNativeConfig-test.ts (hunk 1 above)
Comment (Kudo) [reply packages/expo-modules-autolinking/src/reactNativeConfig/__tests__/reactNativeConfig-test.ts:166]: haha i'm leaning toward to keep test case names as lowercases 😅

**Comment by Kudo** [comment]
### Merge activity

* **Aug 9, 9:04 AM EDT**: @Kudo started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/expo/expo/30913).

**Pull request merged**